### PR TITLE
affiliations: align with custom vocabulary backend

### PR DESCRIPTION
**Pull request by zzacharo**
requires https://github.com/inveniosoftware/react-invenio-forms/pull/82
closes https://github.com/inveniosoftware/invenio-app-rdm/issues/919

### CODE REVIEW

**Review by ppanero — APPROVED** [review]
Discussed IRL. Approving to unblock, before merge:
- Tests should pass
- `name` should be customizable (e.g. to `subject`) in the `AllowAdditions/VocabularyField`

**Comment by zzacharo** [comment]
This is ready to merged. **NOTE**: I simplified the serialization by allowing passing to the backend both `id` and `name` for the submitted affilliation values! That means that if merged the affiliations won't work until the backend behaviour is changed! For moving things forward we can merge this :)

**Pull request merged**
